web/src/lib/Signer.ts

```diff
@@ -16,7 +19,28 @@ interface Window {
 }
 declare const window: Window;
 
+let bunkerSigner: BunkerSigner | undefined;
+let nip46CachedPublicKey: string | undefined;
+
 export class Signer {
+	public static async establishBunkerConnection(bunker: string): Promise<void> {
+		const bp = await parseBunkerInput(bunker);
+		if (!bp) throw new Error(`failed to parse '${bunker}'`);
+
+		const storage = new WebStorage(localStorage);
+		let kss = storage.get('nip46clientSecret');
+		let ks: Uint8Array;
+		if (kss) {
+			ks = hexToBytes(kss);
+		} else {
+			ks = generateSecretKey();
+			storage.set('nip46clientSecret', bytesToHex(ks));
+		}
+
+		bunkerSigner = new BunkerSigner(ks, bp);
```

Comment:
I'll try with nsec.app. I never try anything with it because every time I try to use it I get some weird browser-related problem, but maybe it is good now. Thank you for trying.